### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by cbartz**
This PR adds the pull request template to be used for our repos, and also adds the src-doc generation mechanism.
The pr template has been adapted because this repo does not contain any charm source code.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+### Module Changes
+
+<!-- Any high level changes to modules and why -->
+
+### Checklist
+
+- [ ] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
+- [ ] The documentation is generated using `src-docs`
+- [ ] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
+
```
Comment (yanksyoon) [inline .github/pull_request_template.md:20]: I believe it isn't in the ISD062 - Pull Request Creation doc but I have further added a few items, that the list becomes 
```
- [ ] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [ ] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [ ] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [ ] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated.
- [ ] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
```

**Review by yanksyoon — DISMISSED** [review]
LGTM!

File: .github/pull_request_template.md (hunk 1 above)
Comment (cbartz) [reply .github/pull_request_template.md:20]: Yes, you are right, the last 3 lines are not from ISD062, but from https://github.com/canonical/jenkins-k8s-operator/pull/24, I think. Thanks for the contribution!

 I'll close this conversation, if you want to add more feel free to reopen it.

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Pull request merged**